File: ConcentrationGame/Concentration.py
#include <iostream>

using namespace std;
bool validString(string val) {
    if(val.length() == 1) {
        return true;
    }
    unsigned int i = 0;
    //cout << "Finding Split!" << endl;
    for(i = 0; i < val.length() - 1; i++) {
        if(val.at(i) > val.at(i+1)) {
            break;
        }
    } 
   // cout << "Found it !" << endl;
    if(i == val.length()-1) {
        return true;
    }
    else {
        string start;
        start = val.substr(0, i+1);
      
        //cout << "Start: " << start;
        //cout << "Val: " << val;
        string end = val.substr(i+1);
            if(start.length() < end.length()) {
                return true;
            } else if (start.length() == end.length()) {
                auto i = start.begin();
                auto j = end.begin();
                while (i != start.end()) {
                    if (*i > *j) {
                        return false;
                    }
                }
                return true;
        } else {
            return false;
        }
    }
}

int main() {
    int num;
    string val;
    cin >> num;
    for(int i = 0; i < num; i++) {
        cin >> val;
        if(validString(val)) {
            cout << '1';
        } else {
            cout << '0';
        }
    }
}

File: ConcentrationGame/Concentration.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

int main() {

	int n = 0;
	int a = 0;
	int b = 0;
	int found = 0;
	int aVal;
	int bVal;
	cin >> n;
	int* valArray = new int[n+1];
	a = 1;
	b = 2;
	aVal = 0;
	bVal = 0;
	for(int i = 1; i <= n; i++) {
		valArray[i] = -1;
	}
	printf("%d %d", a, b);
	//fflush(stdout);
	do {
	    fflush(stdout);
		cin >> aVal;
		if(cin.fail()) {
			found++;
			cin.clear();
			cin.ignore();
			//fflush(stdout);
			//fflush(stdin);
			continue;

		}
		if(aVal == -1) {
			break;
		} else {
			if(valArray[aVal] != -1) {
				//cout << a << " " << valArray[aVal];
				//cout << valArray[aVal] << " " << a;
				//fflush(stdout);
				printf("%d %d", a, valArray[aVal]);
                continue;
			} else {
				valArray[aVal] = a;
			}
			cin >> bVal;
			if(valArray[bVal] != -1) {
				printf("%d %d", b, valArray[bVal]);
				continue;
			} else {
				valArray[bVal] = b;
			}
		}
		a+=2;
		b+=2;
		printf("%d %d", a, b);
		//fflush(stdout);
	} while(found < n);
	cout << "-1";
	//fflush(stdout);
	return 0;
}